larky/src/main/resources/vendor/Crypto/OpenPGP/__init__.star

```diff
@@ -0,0 +1,265 @@
+
+StopIterating = Error("StopIteration")
+
+def _ensure_bytes(n, chunk, g):
+    for _while_ in range(WHILE_LOOP_EMULATION_ITERATION):
+        if len(chunk) >= n:
+            break
+        next_chunk = g.next()
+        if next_chunk == StopIterating:
+            return fail("OpenPGPException: Not enough bytes")
+        # chunk += next(g)
+        chunk += next_chunk
+    return chunk
+
+def _gen_one(seq):
+    self = larky.mutablestruct(__class__='generator')
+    def __init__(seq):
+        self._sequence = seq
+        self._length = len(seq)
+        self._current = -1
+        return self
+    self = __init__(seq)
+
+    def next():
+        if self._current >= self._length-1:
+            return StopIterating
+        self._current += 1
+        return self._sequence[self._current]
+    self.next = next
+    return self
+
+def _slurp(g):
+    # bs = b''
+    s = ''
+    for chunk in g:
+        s += tostr(chunk)
+    return tobytes(s)
+
+def PushbackGenerator(g):
+    self = larky.mutablestruct(__name__='PushbackGenerator', __class__=PushbackGenerator)
+    def __init__(g):
+        # self._g = g
+        self._g = g
+        self._pushback = []
+        return self
+    self = __init__(g)
+
+    def __iter__():
+        return self
+    self.__iter__ = __iter__
+
+    def next():
+        return self.__next__()
+    self.next = next
+
+    def __next__():
+        if len(self._pushback):
+            return self._pushback.pop(0)
+        # return next(self._g)
+        return self._g.next()
+    self.__next__ = __next__
+
+    def hasNext():
+        if len(self._pushback) > 0:
+            return True
+        # try:
+        #     chunk = next(self)
+        #     self.push(chunk)
+        #     return True
+        # except StopIteration:
+        #     return False
+        chunk = self.next()
+        if chunk ==  StopIterating:
+           return False
+        self.push(chunk)
+        return True
+
+    self.hasNext = hasNext
+
+    def push(i):
+        if hasattr(self._g, 'push'):
+            self._g.push(i)
+        else:
+            self._pushback.insert(0, i)
+    self.push = push
+    return self
+
+    
+def Packet():
+    """ OpenPGP packet.
+        http://tools.ietf.org/html/rfc4880#section-4.1
+        http://tools.ietf.org/html/rfc4880#section-4.3
+    """
+
+    self = larky.mutablestruct(__name__='Packet', __class__=Packet)
+
+    def __init__(data=None):
+        for tag in Packet.tags:
+            if Packet.tags[tag] == self.__class__:
+                self.tag = tag
+                break
+        self.data = data
+    self = __init__(data)
+
+    def parse(input_data):
+        if hasattr(input_data, 'next') or hasattr(input_data, '__next__'):
+            g = PushbackGenerator(input_data)
+        else:
+            g = PushbackGenerator(_gen_one(input_data))
+
+        packet = None
+        # If there is not even one byte, then there is no packet at all
+        # chunk = _ensure_bytes(1, next(g), g)
```

Comment:
FYI - next() and iter() now work in Larky. https://github.com/verygoodsecurity/starlarky/pull/161

https://github.com/verygoodsecurity/starlarky/blob/ceb142820d497baeabd9fde4b58001212904e280/larky/src/test/resources/stdlib_tests/test_builtins.star#L167

Reply:
That's so cool! will update it